### PR TITLE
Adjust style of passwordless login

### PR DESCRIPTION
**Pull request by susnux**
* Follow up on #38131 

## Summary
This aligns the style of passwordless login with the current reworked login box:
* Use `NcTextInput` for username (so it now matches visually with the login box (e.g. look at the border on the screenshots))
* Remove background of error messages
* Use a button for *back* instead of a link
* Add a heading so you (as a user) on which form you are

### Screenshots
before|after
---|---
image|image
image|image
.|image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Screenshots before/after for front-end changes
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes) (*none as this is introduced only for NC27+*)

### CODE REVIEW

File: core/src/views/Login.vue [hunk 1]
```diff
@@ -73,9 +73,13 @@
 						:is-localhost="isLocalhost"
 						:has-public-key-credential="hasPublicKeyCredential"
 						@submit="loading = true" />
-					<a href="#" class="login-box__link" @click.prevent="passwordlessLogin = false">
-						{{ t('core', 'Back') }}
-					</a>
+					<NcButton type="tertiary"
+						style="margin-top: .5rem;"
```
Comment (Pytal) [inline core/src/views/Login.vue:77]: Move into `style` block?

File: core/src/views/Login.vue [hunk 2]
```diff
@@ -73,9 +73,13 @@
...
+						:aria-label="t('core', 'Back to login form')"
+						:wide="true"
+						@click="passwordlessLogin = false">
+						Back
```
Comment (Pytal) [inline core/src/views/Login.vue:81]:
```suggestion
						{{ t('core', 'Back') }}
```

File: core/src/components/login/PasswordLessLoginForm.vue [hunk 3]
```diff
@@ -99,6 +94,9 @@ export default {
 	},
 	methods: {
 		authenticate() {
+			// check required fields
+			if (!this.$refs.loginForm.checkValidity()) return
```
Comment (Pytal) [inline core/src/components/login/PasswordLessLoginForm.vue:98]:
```suggestion
			if (!this.$refs.loginForm.checkValidity()) {
				return
			}
```

**Comment by susnux** [comment]
@Pytal thank you for your review, fixed your comments :smiley:

**Review by skjnldsv — APPROVED** [review]

**Review by jancborchardt — APPROVED** [review]
Very nice @susnux, looks good design-wise! :)

**Pull request merged**